src/solidClientHelpers/index.ts

```diff
@@ -506,3 +520,184 @@ export function getThirdPartyPermissions(
+    (a) => unstable_setAgentDefaultAccess(a as AclDataset, webId, access)
+  );
+
+  return aclDataset as AclDataset;
+}
+
+export function vcardExtras(property: string): string {
+  return `http://www.w3.org/2006/vcard/ns#${property}`;
+}
+
+interface IContent {
+  iri: string;
+  dataset: SolidDataset;
+}
+
+interface IAddressBookContent {
+  index: IContent;
+  groups: IContent;
+  people: IContent;
+}
+
+interface ICreateAddressBook {
+  iri: string;
+  owner: string;
+  title: string;
+}
+
+export function createAddressBook({
+  iri,
+  owner,
+  title = "Contacts",
+}: ICreateAddressBook): IAddressBookContent {
+  const createIri = (path: string): string => `${iri}/${path}`;
+  const indexIri = createIri("index.ttl");
+  const peopleIri = createIri("people.ttl");
+  const groupsIri = createIri("groups.ttl");
+
+  const groups = defineDataset();
+  const people = defineDataset();
+  const index = defineDataset(
+    (t) => addUrl(t, rdf.type, vcardExtras("AddressBook")),
+    (t) => addUrl(t, acl.owner, owner),
+    (t) => addStringUnlocalized(t, dc.title, title),
+    (t) => addUrl(t, vcardExtras("nameEmailIndex"), peopleIri),
+    (t) => addUrl(t, vcardExtras("groupIndex"), groupsIri)
+  );
+
+  return {
+    index: {
+      iri: indexIri,
+      dataset: index,
+    },
+    groups: {
+      iri: groupsIri,
+      dataset: groups,
+    },
+    people: {
+      iri: peopleIri,
+      dataset: people,
+    },
+  };
+}
+
+export async function getAddressBook(
+  iri: string,
+  fetch: (req: Request) => Promise<void>
+): Promise<IResponse> {
+  const { respond, error } = createResponder();
+
+  try {
+    const indexIri = `${iri}/index.ttl`;
+    const index = await unstable_fetchLitDatasetWithAcl(indexIri, { fetch });
+    const type = getUrl(index, rdf.type);
+
+    if (!type || type !== vcardExtras("AddressBook")) {
+      throw new Error(`${iri} is not an AddressBook`);
+    }
+
+    const groupsIri = getUrl(index, vcardExtras("groupIndex"));
+    const peopleIri = getUrl(index, vcardExtras("nameEmailIndex"));
+
+    const groups = await unstable_fetchLitDatasetWithAcl(groupsIri as string, {
+      fetch,
+    });
+    const people = await unstable_fetchLitDatasetWithAcl(peopleIri as string, {
+      fetch,
+    });
+
+    const response = {
+      index,
+      groups,
+      people,
+    };
+
+    return respond(response);
+  } catch ({ message }) {
+    return error(message);
+  }
+}
+
+export interface ISaveNewAddressBook {
+  iri: string;
+  owner: string;
+  title: string;
+  fetch: (request: Request) => Promise<void>;
+}
+
+export async function saveNewAddressBook({
+  iri,
+  owner,
+  title = "Contacts",
+  fetch,
+}: ISaveNewAddressBook): Promise<IResponse> {
+  const { respond, error } = createResponder();
+  const { response, error: message } = await getAddressBook(iri, fetch);
+
+  if (response) return respond(response);
+
+  if (message && message.match(/4\d{2}/)) {
```

Comment:
Couldn't this be problematic, e.g. 403 (Forbidden) would still result in trying to create an Address Book, correct?

Reply:
I think this should be specific for 404?